File: Prime Numbers/Range Of Prime/Sieve Of Eratosthenes/SOE.c
#include <stdio.h>
#include <stdlib.h>

// Sieve of Eratosthenes
char *seive(int max) {
	// Prime list
	char *p = (char *)malloc((max+1)*sizeof(char));

	// Set Value to true
	for (int i=2; i<=max; i++) { p[i] = '1'; }

	p[0] = p[1] = '0';
	for (int i=2; i*i<=max; i++)
		if ( p[i]=='1' )
			for (int j=i*i; j<=max; j+=i)
				p[j] = '0';

	return p;
}

// DRIVER CODE
void main() {
	// Upmost limit
	int max;
	scanf("%d", &max);

	// Get Primelist
	char *primeList = seive(max);

	// Print prime list
	for ( int i=0; i<=max; i++ )
		if ( primeList[i]=='1' )
			printf("%d ", i);
}

File: Prime Numbers/Range Of Prime/Sieve Of Eratosthenes/SOE.java
import java.util.ArrayList;
import java.util.Scanner;

public class SOE {
	// Seive OF Eratosthenes
	private static ArrayList<Integer> seive(int max) {
		// Bool List
		boolean[] p = new boolean[max+1];
		for (int i=2; i<=max; i++) { p[i]=true; }

		// Get primes
		for (int i=2; i*i<=max; i++)
			if ( p[i] )
				for (int j=i*i; j<=max; j+=i)
					p[j] = false;

		// Get only primes
		ArrayList<Integer> prime = new ArrayList<>();
		for (int i=0; i<=max; i++)
			if ( p[i] )
				prime.add(i);

		return prime;
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);

		// Get max limit
		int max= sc.nextInt();

		// Get prime list
		ArrayList<Integer> primeList = seive(max);

		// Print list
		System.out.println(primeList);
	}
}